codegen/smithy-dafny-codegen/src/main/java/software/amazon/smithy/smoketests/traits/SmokeTestCaseValidator.java

```diff
@@ -0,0 +1,107 @@
+// Copyright Amazon.com Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+// This is a forked copy of
+// https://github.com/smithy-lang/smithy/blob/main/smithy-smoke-test-traits/src/main/java/software/amazon/smithy/smoketests/traits/SmokeTestCaseValidator.java
+// with a piece of overly-restrictive validation turned off
+// so that we can use smoke tests to test input validation as well.
+// See https://github.com/smithy-lang/smithy/issues/2282
+
+package software.amazon.smithy.smoketests.traits;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import software.amazon.smithy.model.Model;
+import software.amazon.smithy.model.knowledge.OperationIndex;
+import software.amazon.smithy.model.node.Node;
+import software.amazon.smithy.model.node.ObjectNode;
+import software.amazon.smithy.model.shapes.Shape;
+import software.amazon.smithy.model.shapes.ShapeId;
+import software.amazon.smithy.model.shapes.StructureShape;
+import software.amazon.smithy.model.validation.AbstractValidator;
+import software.amazon.smithy.model.validation.NodeValidationVisitor;
+import software.amazon.smithy.model.validation.ValidationEvent;
+import software.amazon.smithy.model.validation.node.TimestampValidationStrategy;
+
+/**
+ * Validates the following:
+ *
+ * <ul>
+ *     <li>vendorParamsShape is a valid shape.</li>
+ *     <li>vendorParams is compatible with vendorParamsShape.</li>
+ *     <li>input is valid for the operation under test.</li>
+ * </ul>
+ */
+public class SmokeTestCaseValidator extends AbstractValidator {
```

Comment:
Question: Do you want to call this `PolymorphSmokeTestCaseValidator` or `SmithyDafnySmokeTestCaseValidator`?

I get why you forked the logic,
but forked logic with the exact same name & class path 
frightens me.

Fork with intention.

Reply:
Unfortunately this only works because I'm using the same name, so that this class gets picked up instead of the other one.

It's fairly yucky monkey patching I'm only okay with because if it fails our smokeTests will become invalid, so the failure is loud and obvious.

Reply:
Damn.